src/main/java/codeu/controller/RegisterServlet.java

```diff
@@ -17,4 +24,62 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     // Forward request to register.jsp file
     request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
   }//doGet
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+    throws IOException, ServletException 
+  {
+    String username = request.getParameter("username");
+    String password = request.getParameter("password");
+    // Encrypt password
+    String passwordHash = BCrypt.hashpw(password, BCrypt.gensalt());
+		
+		// Username Error Handling
+    if (!username.matches("[\\w*\\s*]*")) 
+    {
+      request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
+      request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
+      return;
+		}//if
+
```

Comment:
The formatting for the bad username should be changed so the curly braces are aligned and the comment aligned to the top of the if statement